### ISSUE TITLE
decodeFirstSync does not throw error for unused bytes

### ISSUE THREAD

**Issue by ashisherc** [hildjj/node-cbor#111]
```
try {
cbor.decodeFirst("01f3d81803d9b0ac6a56004d5d6117eed4857b95c33a6de2c7bcc234492f2b32409e0aa015d2407ab37dd671c21ea7a189d5a9163abc30d107");
  } catch (error) {
    // no error
  }
```

while using cbor.decodeFirst does throw error about invalid f3 after 01,

The expected behaviour is to throw an error to be able to catch in the catch block.

**Comment by hildjj** [comment]
I agree that decodeFirst and decodeFirstSync should do the same thing.  This is an edge-case breaking change, so I'll update the minor version number in response.

**Comment by hildjj** [comment]
Fixed in v5.1.0
